pyccel/ast/builtins.py

```diff
@@ -296,8 +296,7 @@ def __new__(cls, arg0, arg1=LiteralFloat(0)):
         return super().__new__(cls)
 
     def __init__(self, arg0, arg1 = LiteralFloat(0)):
-        self._is_cast = arg0.dtype is NativeComplex() and \
-                        isinstance(arg1, Literal) and arg1.python_value == 0
+        self._is_cast = isinstance(arg1, Literal) and arg1.python_value == 0
```

Comment:
What is the exact meaning of the attribute `_is_cast`?

It will be `False` if I try to convert a complex number into another one with different precision.

Reply:
If you try to convert a complex number into another one with different precision then you will have:
```
arg0 = my_complex_object,
arg1 = LiteralFloat(0)
```
so `_is_cast` will be true.

`_is_cast` simply indicates that the function was called as a cast rather than as a function assembling 2 floats/integers into a complex number.

Before this PR `_is_cast` indicated if a complex was being cast. I am not sure where this extra restriction came from. It meant that `np.complex(1+2j)` would be treated as a cast while `complex(1)` would be treated the same as `complex(1, 0)`

Reply:
Probably the aim was just to reduce the number of cases but if `complex(1)` is treated the same as `complex(1, 0)` then `complex(np.ones(4))` is also treated the same as `complex(np.ones(4), 0)`. This is exactly the case that doesn't work with the intel compiler bug (it also looks harder to treat in C, but I think the unravelling must handle it)

Reply:
Thanks for the answer. We should have sufficient tests to catch any errors here.

Reply:
Indeed my first attempt at correcting this line was wrong and was caught :sweat_smile: